src/HasBinaryUuid.php

```diff
@@ -73,17 +73,90 @@ public static function decodeUuid(string $binaryUuid): string
 
     public function toArray()
     {
-        if (! $this->exists) {
-            return parent::toArray();
+        $uuidAttributes = $this->getUuidAttributes();
+
+        $array = parent::toArray();
+
+        // if (! $this->exists) {
+        // return $array;
+        // }
+
+        if (is_array($uuidAttributes)) {
+            foreach ($uuidAttributes as $attributeKey) {
+                if (array_key_exists($attributeKey, $array)) {
+                    $uuidKey = $this->getRelatedBinaryKeyName($attributeKey);
+                    $array[$attributeKey] = $this->{$uuidKey};
+                }
+            }
+        }
+
+        return $array;
+    }
+
+    public function getRelatedBinaryKeyName($attrib)
+    {
+        $suffix = $this->getUuidTextAttributeSuffix();
+
+        return "{$attrib}{$suffix}";
+    }
+
+    public function getAttribute($key)
+    {
+        if (($uuidKey = $this->uuidTextAttribute($key)) && $this->{$uuidKey} !== null) {
+            return static::decodeUuid($this->{$uuidKey});
+        }
+
+        return parent::getAttribute($key);
+    }
+
+    public function setAttribute($key, $value)
+    {
+        if ($uuidKey = $this->uuidTextAttribute($key)) {
+            $value = static::encodeUuid($value);
         }
 
-        $data = parent::toArray();
+        return parent::setAttribute($key, $value);
+    }
+
+    protected function getUuidTextAttributeSuffix()
+    {
+        return (property_exists($this, 'uuidTextAttribSuffix')) ? $this->uuidTextAttribSuffix : '_text';
+    }
+
+    protected function uuidTextAttribute($key)
```

Comment:
Do I understand this code correctly that when calling `$model->uuid_text`, the actual property's name is calculated, so the return value would actually be the textual version value of the primary key? You're adding this as a shortcut, I suppose?

Reply:
Not a shortcut per say. What i'm doing here is to avoid hard coding `->uuid_text` to the property name for textual UUIDs and to also make it customizable to that developers can choose another property name suffix apart from `'_text'`

Reply:
yeah, that's fine.